src/cashlib/cashlib.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2015-2018 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+/* clang-format off */
+// must be first for windows
+#include "compat.h"
+
+/* clang-format on */
+#include "base58.h"
+#include "random.h"
+#include "streams.h"
+#include "utilstrencodings.h"
+
+#include <openssl/rand.h>
+#include <string>
+#include <vector>
+
+static bool sigInited = false;
+
+// stop the logging
+int LogPrintStr(const std::string &str) { return str.size(); }
+namespace Logging
+{
+uint64_t categoriesEnabled = 0; // 64 bit log id mask.
+};
+
+// helper functions
+namespace
+{
+#if 0
+CKey LoadKey(uint256 val)
+{
+    CKey secret;
+    unsigned char *src = val.begin();
+    secret.Set(src, src+32, true);
+    return secret;
+}
+#endif
+
+CKey LoadKey(unsigned char *src)
+{
+    CKey secret;
+    secret.Set(src, src + 32, true);
+    return secret;
+}
+
+// From core_read.cpp #include "core_io.h"
+bool DecodeHexTx(CTransaction &tx, const std::string &strHexTx)
```

Comment:
why redefining `DecodeHexTx`? is already defined in ` core_read.cpp ` (as you noted already in the comment. doesn't his introduce code duplication?

Reply:
cashlib.cpp only becomes libbitcoincash.so which is not linked into bitcoind.  There were other symbols in core_read.cpp that was causing libbitcoincash.so to pull in other .cpp recursively so I had to pull this out.